comp110-worksheet-6/OxoBoard.cs

```diff
@@ -3,74 +3,149 @@
-					}
-				}
-
-				Console.WriteLine();
-			}
-		}
-	}
+    public enum Mark { None, O, X };
+
+    public class OxoBoard
+    {
+
+        public int width;
+        public int height;
+        public Regex format = new Regex("^[0-2](,[0-2])+$"); //Regular expression that will only allow coma separated integers (0 to 2) ex. 0,0; 0,1; 1,1; 0,2
+
+        public Mark[,] gameBoard;
+
+        // Constructor. Perform any necessary data initialisation here.
+        // Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
+        public OxoBoard(/* int width = 3, int height = 3, int inARow = 3 */)
+        {
+            width = height = 3;
+            gameBoard = new Mark[width, height];
+
+            for (int i = 0; i < width; i++)
+            {
+                for (int j = 0; j < height; j++)
+                {
+                    gameBoard[i, j] = Mark.None;
+                }
+            }
+        }
+
+        // Return the contents of the specified square.
+        public Mark GetSquare(int x, int y)
+        {
+            return gameBoard[x, y];
+        }
+
+        // If the specified square is currently empty, fill it with mark and return true.
+        // If the square is not empty, leave it as-is and return False.
+        public bool SetSquare(int x, int y, Mark mark)
+        {
+            string inputString = x + "," + y;
+            if (format.IsMatch(inputString))
+            {
+                if (gameBoard[x, y] == Mark.None)
+                {
+                    gameBoard[x, y] = mark;
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        // If there are still empty squares on the board, return false.
+        // If there are no empty squares, return true.
+        public bool IsBoardFull()
+        {
+            for (int i = 0; i < width; i++)
+            {
+                for (int j = 0; j < height; j++)
+                {
+                    if (gameBoard[i, j] == Mark.None)
+                    {
+                        return false;
+                    }
+                }
+            }
+            return true;
+        }
+
+        // If a player has three in a row, return Mark.O or Mark.X depending on which player.
+        // Otherwise, return Mark.None.
+        public Mark GetWinner()
+        {
+            for (int i = 0; i < width; i++)
+            {
+                for (int j = 0; j < height; j++)
+                {
+                    if (gameBoard[i, j] != Mark.None) //Only look inside the cell if it's not empty.
+                    {
+                        int currI = i + 2, currJ = j + 2;
+                        int[] currSquare = new int[] { i, j };
+                        if (currI < 3 && currJ < 3)
```

Comment:
I suspect this check is what's causing some of the unit tests to fail -- some of the tests are being skipped